### ISSUE TITLE
Programming language agnostic API?

### ISSUE THREAD

**Issue by adibrastegarnia** [p4lang/p4-constraints#68]
I am wondering if this API can be programming language agnostic or somehow auto generated for different programming languages?  What is the best way to implement the library for a programming language like Go?

**Comment by adibrastegarnia** [comment]
@smolkaj  yeah this is one of the solutions that I have in my mind which is a good starting point but potentially could raise concerns about adding one hop if it is supposed to be integrated by a controller.  Auto generating Validate function in different programming languages is the best option but not sure how easy is that to achieve in this context (e.g. https://github.com/envoyproxy/protoc-gen-validate)

**Comment by adibrastegarnia** [comment]
@smolkaj 
Can we expect anything soon that addresses this issue?

**Comment by smolkaj** [comment]
I am not planning to work on this currently, but contributions are of course very welcome.
